src/util/log.h

```diff
@@ -93,58 +113,96 @@ void print_stacktrace(unsigned int max_frames = 63);
-    _smek_log(buffer, LogLevel::INFO);
+    len += sntprint(print_buffer, LOG_BUFFER_SIZE, WHITE "I {}" RESET " @ {} ({}): ", file, line, func);
+    len += sntprint(print_buffer + len, LOG_BUFFER_SIZE - len, message_buffer);
+    len += sntprint(print_buffer + len, LOG_BUFFER_SIZE - len, "\n");
+    LogMessage log_message = {
+        file,
+        line,
+        func,
+    };
+    std::memcpy(&log_message.message, message_buffer, LOG_BUFFER_SIZE);
+    _smek_log(print_buffer, print_buffer, LogLevel::INFO, log_message);
 }
 
 template <typename... Args>
 void _smek_log_trace(const char *file, u32 line, const char *func, const char *message, Args... args) {
-    char buffer[LOG_BUFFER_SIZE] = {};
+    char message_buffer[LOG_BUFFER_SIZE] = {};
+    char print_buffer[LOG_BUFFER_SIZE] = {};
+    sntprint(message_buffer, LOG_BUFFER_SIZE, message, args...);
     u32 len = 0;
-    len += sntprint(buffer, LOG_BUFFER_SIZE, "T {} @ {} ({}): ", file, line, func);
-    len += sntprint(buffer + len, LOG_BUFFER_SIZE - len, message, args...);
-    len += sntprint(buffer + len, LOG_BUFFER_SIZE - len, "\n");
-    _smek_log(buffer, LogLevel::TRACE);
+    len += sntprint(print_buffer, LOG_BUFFER_SIZE, "T {} @ {} ({}): ", file, line, func);
+    len += sntprint(print_buffer + len, LOG_BUFFER_SIZE - len, message_buffer);
+    len += sntprint(print_buffer + len, LOG_BUFFER_SIZE - len, "\n");
+    LogMessage log_message = {
+        file,
+        line,
+        func,
+    };
+    std::memcpy(&log_message.message, message_buffer, LOG_BUFFER_SIZE);
+    _smek_log(print_buffer, print_buffer, LogLevel::TRACE, log_message);
 }
 
 template <typename... Args>
 void _smek_unreachable(const char *file, u32 line, const char *func, const char *message, Args... args) {
-    char buffer[LOG_BUFFER_SIZE] = {};
+    char message_buffer[LOG_BUFFER_SIZE] = {};
+    char print_buffer[LOG_BUFFER_SIZE] = {};
+    sntprint(message_buffer, LOG_BUFFER_SIZE, message, args...);
     u32 len = 0;
-    len += sntprint(buffer, LOG_BUFFER_SIZE, RED "U {}" RESET " @ {} ({}) unreachable:\n", file, line, func);
-    len += sntprint(buffer + len, LOG_BUFFER_SIZE - len, BOLDRED "| " RESET);
-    len += sntprint(buffer + len, LOG_BUFFER_SIZE - len, message, args...);
-    len += sntprint(buffer + len, LOG_BUFFER_SIZE - len, "\n" BOLDRED "|" RESET " Stacktrace:\n");
-    _smek_log(buffer, LogLevel::ERROR);
+    len += sntprint(print_buffer, LOG_BUFFER_SIZE, RED "U {}" RESET " @ {} ({}) unreachable:\n", file, line, func);
+    len += sntprint(print_buffer + len, LOG_BUFFER_SIZE - len, BOLDRED "| " RESET);
+    len += sntprint(print_buffer + len, LOG_BUFFER_SIZE - len, message_buffer);
+    len += sntprint(print_buffer + len, LOG_BUFFER_SIZE - len, "\n" BOLDRED "|" RESET " Stacktrace:\n");
+    LogMessage log_message = {
+        file,
+        line,
+        func,
+    };
+    std::memcpy(&log_message.message, message_buffer, LOG_BUFFER_SIZE);
```

Comment:
Feel like this can be a function? Maybe throw in "_smek_log" in the function aswell?